app/plugin_requests.go

```diff
@@ -67,19 +67,30 @@ func (a *App) servePluginRequest(w http.ResponseWriter, r *http.Request, handler
 		session, err := a.GetSession(token)
 		csrfCheckPassed := true
 
-		if err == nil && cookieAuth && r.Method != "GET" && r.Header.Get(model.HEADER_REQUESTED_WITH) != model.HEADER_REQUESTED_WITH_XML {
-			bodyBytes, _ := ioutil.ReadAll(r.Body)
-			r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
-			r.ParseForm()
-			sentToken := r.FormValue("csrf")
+		if err == nil && cookieAuth && r.Method != "GET" {
+			sentToken := ""
+
+			if r.Header.Get(model.HEADER_CSRF_TOKEN) == "" {
+				bodyBytes, _ := ioutil.ReadAll(r.Body)
+				r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+				r.ParseForm()
+				sentToken = r.FormValue("csrf")
+				r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+			} else {
+				sentToken = r.Header.Get(model.HEADER_CSRF_TOKEN)
+			}
+
 			expectedToken := session.GetCSRF()
 
 			if sentToken != expectedToken {
 				csrfCheckPassed = false
 			}
 
-			// Set Request Body again, since otherwise form values aren't accessible in plugin handler
-			r.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+			// ToDo(DSchalla) 2019/01/04: Remove after deprecation period and only allow CSRF Header
```

Comment:
The typical process here is to make the follow up ticket and include the ticket number in the comment.

Reply:
TicketID added